### ISSUE TITLE
GraphQL cart items price currency

### ISSUE THREAD

**Issue by leroynas** [magento/magento2#29041]
The currency of the Money object in the cart query are not corresponding to the either Content-Currency and base currency. The value of the price is updated but the currency is not.

### Preconditions (*)

1. Magento 2.4-develop

### Steps to reproduce (*)

1. Setup a Magento store with multiple currencies on the default store view
2. Create cart (graphql)
3. Add product to cart (graphql)
4. Get cart (graphql) - make sure to add non default currency code as Header: Content-Currency (USD is used in example)

```
query {
  cart(cart_id: "nCI33c1lc0JUETLzvQvliMLt9Lpy7URy") {
    items {
      prices {
        row_total {
          value
          currency
        }
      }
    }
  }
}
```

### Expected result (*)

1. Currency property on Money object should return the chosen currency
```
{
    "data": {
        "cart": {
            "items": [
                {
                    "prices": {
                        "price": {
                            "value": 31.48,
                            "currency": "USD"
                        }
                    }
                }
            ]
        }
    }
}
```

### Actual result (*)

1. Currency will always be the default currency
```
{
    "data": {
        "cart": {
            "items": [
                {
                    "prices": {
                        "price": {
                            "value": 31.48,          <- this value is in US dollar
                            "currency": "EUR"        <- non corresponding currency
                        }
                    }
                }
            ]
        }
    }
}
```

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by leroynas** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @leroynas. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by leroynas** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @leroynas. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @leroynas, here is your [Magento instance](https://i-29041-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-29041-2-4-develop.instances.magento-community.engineering/admin_4a6c](https://i-29041-2-4-develop.instances.magento-community.engineering/admin_4a6c)
Login: `4684fc4e` Password: `813f0fa79c6f`
Instance will be terminated in up to 3 hours.

**Comment by leroynas** [comment]
This issue is exactly the same on the Magento instance created by the bot:

Content-Currency: USD
```
{
    "data": {
        "cart": {
            "items": [
                {
                    "prices": {
                        "row_total": {
                            "value": 9.21,
                            "currency": "USD"
                        }
                    }
                }
            ]
        }
    }
}
```

Content-Currency: EUR
```
{
    "data": {
        "cart": {
            "items": [
                {
                    "prices": {
                        "row_total": {
                            "value": 6.51,
                            "currency": "USD"
                        }
                    }
                }
            ]
        }
    }
}
```

**Comment by maheshmakwana-code** [comment]
**app\code\VendoreName\ModuleName\etc\graphql\di.xml**
```
<?xml version="1.0" ?>

<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
    <preference for="Magento\QuoteGraphQl\Model\Resolver\CartItemPrices"
                type="VendoreName\ModuleName\Model\Resolver\CartItemPricesUpdate"/>
</config>
```
**app\code\VendoreName\ModuleName\Model\Resolver\CartItemPricesUpdate.php**

```
<?php

namespace VendoreName\ModuleName\Model\Resolver;

use Magento\Framework\Exception\LocalizedException;
use Magento\Framework\GraphQl\Config\Element\Field;
use Magento\Framework\GraphQl\Query\ResolverInterface;
use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
use Magento\Framework\Pricing\PriceCurrencyInterface;
use Magento\Quote\Model\Cart\Totals;
use Magento\Quote\Model\Quote\Item;
use Magento\Quote\Model\Quote\TotalsCollector;
use Magento\Store\Model\StoreManagerInterface;

class CartItemPricesUpdate implements ResolverInterface
{
    private $totalsCollector;
    private $totals;
    private $_storeManager;
    private $priceCurrencyObj;

    public function __construct(
        TotalsCollector $totalsCollector,
        StoreManagerInterface $storeManager,
        PriceCurrencyInterface $priceCurrencyObj
    ) {
        $this->_storeManager = $storeManager;
        $this->priceCurrencyObj = $priceCurrencyObj;
        $this->totalsCollector = $totalsCollector;
    }

    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
    {
        if (!isset($value['model'])) {
            throw new LocalizedException(__('"model" value should be specified'));
        }
        /** @var Item $cartItem */
        $cartItem = $value['model'];

        if (!$this->totals) {
            // The totals calculation is based on quote address.
            // But the totals should be calculated even if no address is set
            $this->totals = $this->totalsCollector->collectQuoteTotals($cartItem->getQuote());
        }
        //$currencyCode = $cartItem->getQuote()->getQuoteCurrencyCode();
        $currencyCode = $this->getCurrentCurrency();

        // update price
        $cartPrice = $this->setConvertPrice($cartItem->getPrice());

        return [
            'price' => [
                'currency' => $currencyCode,
                'value' => $cartPrice,
            ],
            'row_total' => [
                'currency' => $currencyCode,
                'value' => $cartItem->getRowTotal(),
            ],
            'row_total_including_tax' => [
                'currency' => $currencyCode,
                'value' => $cartItem->getRowTotalInclTax(),
            ],
            'total_item_discount' => [
                'currency' => $currencyCode,
                'value' => $cartItem->getDiscountAmount(),
            ],
            'discounts' => $this->getDiscountValues($cartItem, $currencyCode),
        ];
    }

    private function getDiscountValues($cartItem, $currencyCode)
    {
        $itemDiscounts = $cartItem->getExtensionAttributes()->getDiscounts();
        if ($itemDiscounts) {
            $discountValues = [];
            foreach ($itemDiscounts as $value) {
                $discount = [];
                $amount = [];
                /* @var \Magento\SalesRule\Api\Data\DiscountDataInterface $discountData */
                $discountData = $value->getDiscountData();
                $discountAmount = $discountData->getAmount();
                $discount['label'] = $value->getRuleLabel() ?: __('Discount');
                $amount['value'] = $discountAmount;
                $amount['currency'] = $currencyCode;
                $discount['amount'] = $amount;
                $discountValues[] = $discount;
            }
            return $discountValues;
        }
        return null;
    }

    private function getCurrentCurrency()
    {
        return $this->_storeManager->getStore()->getCurrentCurrencyCode();
    }

    public function setConvertPrice($price)
    {
        $storeId = $this->_storeManager->getStore()->getStoreId();
        $currentCurrencyCode = $this->getCurrentCurrency();
        $baseCurrencyCode = $this->_storeManager->getStore()->getBaseCurrencyCode();
        // check current store currency
        if ($baseCurrencyCode == $currentCurrencyCode) {
            return $price;
        } else {
            $price = round($this->priceCurrencyObj->convert($price, $storeId, $currentCurrencyCode), 2);
            return $price;
        }
    }

}

```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-40987` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by engcom-Kilo** [comment]
Hello @leroynas 
Thanks for report. I'm not able to reproduce this issue on latest 2.4-develop branch. Looks like it's been fixed in https://github.com/magento/magento2/commit/797e7f82f84deb09c5e4d519861e52d556f48c66

**Comment by magento-engcom-team** [comment]
Hi @leroynas, @engcom-Kilo. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-40987` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
